File: client-clojure/src/sirqul_io_t_platform/specs/cargo_type.clj
(ns sirqul-io-t-platform.specs.cargo-type
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.retailer :refer :all]
            [sirqul-io-t-platform.specs.service-hub :refer :all]
            )
  (:import (java.io File)))


(def cargo-type-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :business) retailer-spec
   (ds/opt :hub) service-hub-spec
   (ds/opt :name) string?
   (ds/opt :width) float?
   (ds/opt :height) float?
   (ds/opt :depth) float?
   (ds/opt :volume) float?
   (ds/opt :weight) float?
   (ds/opt :loadTime) int?
   (ds/opt :palletizable) boolean?
   (ds/opt :palletRatio) int?
   (ds/opt :adminOnly) boolean?
   })

(def cargo-type-spec
  (ds/spec
    {:name ::cargo-type
     :spec cargo-type-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/CargoType.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CargoType._

case class CargoType (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
business: Option[Retailer],
hub: Option[ServiceHub],
name: Option[String],
width: Option[Double],
height: Option[Double],
depth: Option[Double],
volume: Option[Double],
weight: Option[Double],
loadTime: Option[Long],
palletizable: Option[Boolean],
palletRatio: Option[Integer],
adminOnly: Option[Boolean])

object CargoType {
  import DateTimeCodecs._

  implicit val CargoTypeCodecJson: CodecJson[CargoType] = CodecJson.derive[CargoType]
  implicit val CargoTypeDecoder: EntityDecoder[CargoType] = jsonOf[CargoType]
  implicit val CargoTypeEncoder: EntityEncoder[CargoType] = jsonEncoderOf[CargoType]
}
